prompt/select.ts

```diff
@@ -14,10 +14,15 @@ import { GenericPrompt } from "./_generic_prompt.ts";
 export type SelectKeys = GenericListKeys;
 
 /** Select option options. */
-export type SelectOption = GenericListOption;
+export interface SelectOption extends GenericListOption {
+  options?: SelectValueOptions;
+}
 
 /** Select option settings. */
-export type SelectOptionSettings = GenericListOptionSettings;
+export interface SelectOptionSettings extends GenericListOptionSettings {
+  options: SelectValueSettings;
+  indentLevel: number;
```

Comment:
I messed around with different ways of rendering search results that are nested in categories, and the best solution I came up includes attaching an indentation level to the settings of each value. More details below.